inference-engine/tests/functional/plugin/cpu/single_layer_tests/non_max_suppression.cpp

```diff
@@ -0,0 +1,451 @@
+                 targetDevice) = this->GetParam();
+
+        element::Type paramsPrec, maxBoxPrec, thrPrec;
+        std::tie(paramsPrec, maxBoxPrec, thrPrec) = inPrecisions;
+
+        std::tie(iouThr, scoreThr, softNmsSigma) = thrValues;
+
+        std::vector<ov::Dimension> bounds;
+        std::tie(bounds, targetInDims) = inShapeParams;
+
+        if (!bounds.empty()) {
+            inputDynamicShapes = std::vector<ngraph::PartialShape>{{bounds[BATCHES], bounds[BOXES], 4}, {bounds[BATCHES], bounds[CLASSES], bounds[BOXES]}};
+        } else {
+            size_t batches, boxes, classes;
+            std::tie(batches, boxes, classes) = targetInDims.front();
+            ov::Dimension numBatches(batches), numBoxes(boxes), numClasses(classes);
+            inputDynamicShapes = std::vector<ngraph::PartialShape>{{numBatches, numBoxes, 4}, {numBatches, numClasses, numBoxes}};
+        }
+
+        for (const auto &ts : targetInDims) {
+            size_t numBatches, numBoxes, numClasses;
+            std::tie(numBatches, numBoxes, numClasses) = ts;
+            targetStaticShapes.push_back(std::vector<ngraph::Shape>{{numBatches, numBoxes, 4}, {numBatches, numClasses, numBoxes}});
+            if (maxOutBoxesType == ngraph::helpers::InputLayerType::PARAMETER) {
+                targetStaticShapes.back().push_back(ngraph::Shape{1});
+            }
+        }
+
+        std::shared_ptr<ngraph::Node> maxOutBoxesPerClassNode;
+        auto params = ngraph::builder::makeDynamicParams(paramsPrec, inputDynamicShapes);
+        params[0]->set_friendly_name("param_1");
+        params[1]->set_friendly_name("param_2");
+
+        if (maxOutBoxesType == ngraph::helpers::InputLayerType::PARAMETER) {
+            inputDynamicShapes.push_back(ngraph::PartialShape{1});
+            params.push_back(std::make_shared<ngraph::opset1::Parameter>(element::Type_t::i32, inputDynamicShapes.back()));
+            params[1]->set_friendly_name("param_3");
+            maxOutBoxesPerClassNode = params.back();
+        } else {
+            maxOutBoxesPerClassNode = builder::makeConstant(maxBoxPrec, ngraph::Shape{}, std::vector<int32_t>{maxOutBoxesPerClass});
+        }
+
+        auto iouThrNode = builder::makeConstant(thrPrec, ngraph::Shape{}, std::vector<float>{iouThr})->output(0);
+        auto scoreThrNode = builder::makeConstant(thrPrec, ngraph::Shape{}, std::vector<float>{scoreThr})->output(0);
+        auto softNmsSigmaNode = builder::makeConstant(thrPrec, ngraph::Shape{}, std::vector<float>{softNmsSigma})->output(0);
+        auto nms = std::make_shared<ngraph::op::v5::NonMaxSuppression>(params[0], params[1], maxOutBoxesPerClassNode, iouThrNode, scoreThrNode,
+                                                                       softNmsSigmaNode, boxEncoding, sortResDescend, outType);
+
+        if (targetDevice == CommonTestUtils::DEVICE_CPU) {
```

Comment:
Why do we need this check inside CPU dedicated test?
Please use `makeNgraphFunction` call to make the funtion.